### PR TITLE
Project Files: Recursive folder import (drag-and-drop only)

### LINKED ISSUE #3492: I can't drag folders to the playlist
**Describe the bug:**
When I want to drag a folder with files I cannot and it gives me the symbol the mouse that if it is possible but I try and it does not leave me I have to select all the files and drag them it is very tedious to do it

**Steps to reproduce the behavior:**
1. Go to 'foldes'
2. drag on 'Playlist'
3. nothing happens does not respond

**Expected behavior:**

**System Details:**
- OpenShot Version [2.5.1-dev2
[libopenshot.log](https://github.com/OpenShot/openshot-qt/files/4638663/libopenshot.log)
[openshot-qt.log](https://github.com/OpenShot/openshot-qt/files/4638667/openshot-qt.log)
[openshot-qt.log](https://github.com/OpenShot/openshot-qt/files/4638668/openshot-qt.log)
[libopenshot.log](https://github.com/OpenShot/openshot-qt/files/4638673/libopenshot.log)

]: {{ log_context.openshot_qt_version }}
- Operating System / Distro: Linux Mint 19.3

**Log Files:**
If you are experiencing a crash, please collect and attach logs of the problem.
Log files can be found in your HOME/.openshot_qt/ folder. Please compress or zip
the log files (as they can become quite large).

**Exception / Stacktrace:**
If you are experiencing a crash or strange output on the command line, please
attach a small snippet of the exception here, but please keep it as small as
possible (for readability).

**Screenshots:** (Optional)
If applicable, add screenshots to help explain your problem. You can include screenshots by
copy/pasting them here, dragging-and-dropping into this form, or clicking below and loading
images saved to your computer. All images are public, so please don't post screenshots
containing personal information.

### LINKED ISSUE #3554: Multiple files (drag and drop OR via import dialog) stops when import hits a duplicate file
**Describe the bug**
When importing multiple files, and there is a duplicate already in the Project Files, the import will stop when it hits the duplicate file.  Same behavior whether using Drag and Drop OR using Import Dialog.

**Steps to reproduce the behavior:**
1. Add a few files to Project Files
2. Select multiple files, with a duplicate file not at the end of the list
3. Drag and Drop files into Project Files area
4. There are No error messages but not all the files are copied into the Project Files
5. Same results if use the Import dialog

**Expected behavior**
An error message warning of the duplicate, with option to Skip or Replace File; then continue with the adding process for the rest of the files.

**System Details**
- OpenShot Version: 2.5.1-dev2
- libopenshot Version:  0.2.5-dev2
- Platform: Windows-10-10.0.18362
- Processor: Intel64 Family 6 Model 158 Stepping 10, GenuineIntel
- Machine: Asus Gaming PC with NVidia GPU
- Python version: 3.7.4
- Qt5 version: 5.12.4
- PyQt5 version: 5.12.3
- Qt Detected Languages: ['en-US']
- LANG Environment Variable: 
- LOCALE Environment Variable: 
- Daily Build: Verified issue still exists in daily build: http://github.com/OpenShot/openshot-qt/releases/download/OpenShot-v2.5.1-dev2-1591560328-c29174f1-7cec1822-x86_64.exe

**Log Files**
- [openshot-qt.log](https://s3.amazonaws.com/logs.openshot.org/topfa6wzaw66nhqkto828ftnigf1af15/openshot-qt.zip) (15 KB)
- [libopenshot.log](https://s3.amazonaws.com/logs.openshot.org/topfa6wzaw66nhqkto828ftnigf1af15/libopenshot.zip) (2 KB)

**Exception / Stacktrace**
```No stacktrace found in log files```

**Screenshots** (Optional)
Not necessary as no error message was shown.

### PR DESCRIPTION
**Pull request by ferdnyc**
Not too long ago (#3492), @cpuhead said, "I can't drag folders to the playlist". Well, now you can!

When processing drag-and-drop imports, OpenShot will recurse into any directories it finds, attempting to import all contained media. A few quirks of this mode:

1. The import switches into a "silent running" mode, where the dialog box alerts about files that it can't import are turned off. **Any unrecognized media will just be silently ignored.**
1. **The import is totally flat ­— this doesn't add folders _to_ Project Files**, just lets you import all of the media _inside_ a folder (and possibly subfolders) into the list, which is still as flat as ever.
1. **Image sequences are limited to one per directory** — whether you say Yes or No to the question about importing a sequence, no other sequences will be detected in that same directory for the remainder of the import.
1. As I was testing this, I actually found a few **bugs in my previous image sequence import code, so a lot of that is rewritten** and should be more robust even in the typical non-folder import cases. (That's what led to the new once-per-directory restriction, because it was doing all sorts of messed-up things when it tried to import a second one from the same path.)
1. **I have no _idea_ what will happen if you drag a folder containing an `.osp` file in** with other media, but I imagine it won't be pretty. Just don't do that.
1. In general it's not even remotely bulletproof, nor is it intended to be. Kitchen-sink importing your entire media library on the off chance you might use something is still _not_ a supported way of working with OpenShot, and Project Files is meant to hold only files that are _**actively in use**_ on the Timeline. **It is not a library, it is a workspace.** If you try to do dumb things with this, it will almost certainly break and it'll be your own fault.
1. **The list of files being imported is now sorted** before the import starts, because the results of Python's `os.walk` are all over the place and it was screwing up the image sequence imports to start from weird numbers.
1. I switched over the non-drag-and-drop, dialog-driven Import Files over to using `QUrl` as well, hoping it would enable selection of folders for import that way, too — but at least on Linux with the GNOME Shell file-open dialog, that doesn't work. If you select a folder alone it will descend into it instead of opening it, and if you multiply-select folders and files together it simply refuses to accept the "Open" button click. So, that's a bummer. But I left the `QUrl` changes, because there's some chance it may open up opportunities for importing from more types of shared/network sources, at least on some platforms. I'll be curious to see what it changes, if anything.

Oh, and along the way I fixed the bug @leehockhin reported in #3554, about duplicate import attempts halting the import. Don't know how I missed that one.

Fixes #3492 fixes #3554 (and probably others)

### CODE REVIEW

**Comment by ferdnyc** [comment]
Grrr. Everything was all set, then I went back to fix a few totally preexisting things that got flagged in the code, and a bunch of unrelated stuff snuck in that broke everything. That's what I get for revising on the fly.

**Comment by jonoomph** [comment]
@ferdnyc What is the merge status of this one?

**Comment by jonoomph** [comment]
This looks really cool, I would love to get it merged, if it's ready.

**Comment by jonoomph** [comment]
I resolved the conflicts on this one, but I'll let @ferdnyc merge this one, once you feel it's ready. :+1:

**Comment by ferdnyc** [comment]
@jonoomph 

Good question, I haven't looked at this in MONTHS TBH. I'll take a look after I deal with the timer PR. I agree, it'd be nice to get in there. (Just wish I had a good way of making it also work for File-menu type imports.)

File: src/windows/main_window.py [hunk 1]
```diff
@@ -2084,14 +2089,14 @@ def showDocks(self, docks):
                 dock.show()
 
     def freezeDocks(self):
-        """ Freeze all dockable widgets on the main screen.
-            (prevent them being closed, floated, or moved) """
+        """ Freeze all dockable widgets on the main screen
+            prevent them being closed, floated, or moved) """
```
Comment (ferdnyc) [inline src/windows/main_window.py:2093]:
```suggestion
            (prevent them being closed, floated, or moved) """
```

File: src/windows/models/files_model.py [hunk 2]
```diff
@@ -285,21 +287,19 @@ def add_files(self, files, image_seq_details=None):
                     file_data["media_type"] = "video"
 
                 # Save new file to the project data
-                file = File()
-                file.data = file_data
-
-                if not image_seq_details:
-                    # Try to discover image sequence, if not supplied
-                    image_seq_details = self.get_image_sequence_details(filepath)
+                new_ile = File()
```
Comment (ferdnyc) [inline src/windows/models/files_model.py:290]:
```suggestion
                new_file = File()
```

Dummy.

File: src/windows/models/files_model.py [hunk 3]
```diff
@@ -385,34 +393,68 @@ def get_image_sequence_details(self, file_path):
...
+                self.win.OpenProjectSignal.emit(filepath)
+                return True
+            if os.path.isdir(filepath):
+                import_quietly = True
+                log.info("Recursively importing {}".format(filepath))
+                try:
+                    for r, _, f in os.walk(filepath):
+                        media_paths.extend([
+                            os.path.join(r, p) for p in f
+                            ])
```
Comment (ferdnyc) [inline src/windows/models/files_model.py:442]:
```suggestion
                        media_paths.extend(
                            [os.path.join(r, p) for p in f])
```

Nice to be able to actually SEE it's a list comprehension.

**Comment by ferdnyc** [comment]
Glad I checked this again, caught one dumb typo. Will merge as soon as Travis gives the OK.

**Pull request merged**